ocean_lib/models/erc20_enterprise.py

```diff
@@ -17,15 +17,35 @@ class ERC20Enterprise(ERC20Token):
     CONTRACT_NAME = "ERC20TemplateEnterprise"
 
     def buy_from_fre_and_order(
-        self, order_params: dict, fre_params: dict, from_wallet: Wallet
+        self,
+        order_params: Union[dict, tuple, OrderParams],
+        fre_params: Union[dict, tuple],
+        from_wallet: Wallet,
     ) -> str:
+        # TODO: this will be handled in web3 py
+        provider_fees = None
+        if isinstance(order_params[2], ProviderFees):
+            provider_fees = tuple(order_params[2])
```

Comment:
this is not ok. If OrderParams is a NamedTuple, I'm not sure you can rely on indices. Also, if you set provider fees to None before changing it if it is a ProviderFees type, then will it remain None if it is a tuple?